src/classes/CW_AssociateAudienceRewardHelper.cls

```diff
@@ -0,0 +1,236 @@
+* 0.1 28/05/2014 Alex Panka	  Intial version 
+*/
+//Cristina P: 25/06/14: Changing the class from "with sharing" to "without sharing" 
+//so the user can create related records while cost/brief is in approval
+public without sharing class CW_AssociateAudienceRewardHelper 
+{
+	/* set controller variables */
+	public String 							contextItem {get;set;}
+		
+	public list<CW_Audience_Group__c>       agList{get;set;}
+	public list<CW_Schedule__c>             schList{get;set;}
+	public list<CW_Campaign_Reward__c>      rewList{get;set;}
+	public set<String>                      agsSet=new Set<String>();//agsMap=new map<String,List<Id>>();
+	public map<String,Id>                   agsMapId=new map<String,Id>();
+	public map<String,Id>                   agsrMap=new map<String,Id>();
+	 
+	private set<String>                     newagsset=new set<string>();
+	private set<Id>                         delagsset=new set<Id>();  	
+	private set<String>                     newagsrset=new set<string>();
+	private set<Id>                         delagsrset=new set<Id>();  	
+	private Id                              CampId;
+	
+	public CW_AssociateAudienceRewardHelper(Id CampaignId)
+	{
+		CampId=CampaignId;
+		init();		
+	}
+	
+	public map<String,Boolean> getCheckkAgs()
+	{
+		map<String,Boolean> result=new map<String,Boolean>();
+		for(CW_Campaign_Reward__c rrec:rewList)
+		{
+			for(CW_Audience_Group__c arec:agList)
+			{
+				result.put(''+arec.id+rrec.ID,false);
+			}
+		}
+		for(String key:newagsset){
+			result.put(key,true);
+		}
+		for(String key:agsSet){
+			result.put(key,true);
+		}
+		return result;
+	}
+	
+	public map<String,Boolean> getCheckkAgsr()
+	{
+		map<String,Boolean> result=new map<String,Boolean>();
+		for(CW_Schedule__c srec : schList)
+		{
+			for(CW_Audience_Group__c arec:agList)
+			{
+				for(CW_Campaign_Reward__c rrec:rewList){
+					result.put(''+srec.id+arec.id+rrec.id,false);
+				}
+			}
+		}		
+		for(String key:newagsrset){
+			result.put(key,true);
+		}
+		for(String key:agsrMap.KeySet()){
+			Id recId=agsrMap.get(key);
+			if(!delagsrset.contains(recId))
+				result.put(key,true);
+		}
+		return result;
+	}
+	
+	public void init()
+	{
+		agsMapId=new map<String,Id>();
+		agsrMap=new map<String,Id>();
+			 
+		//Records To create/delete	 
+		newagsset=new set<string>();
+		delagsset=new set<Id>();  	
+		newagsrset=new set<string>();
+		delagsrset=new set<Id>();  	
+		
+		//set the data display mode
+		agList=  [SELECT Id,Name FROM CW_Audience_Group__c  WHERE Campaign__c=:campId AND Is_Control_Group__c=false];
+		schList= [SELECT Id,Name ,Start_Date__c,End_Date__c  FROM CW_Schedule__c WHERE Marketing_Campaign__c=:campId ];
+		rewList= [SELECT Id,Name,Reward_Catalogue__r.Offer_Type__r.Name,Reward_Type__r.Name FROM CW_Campaign_Reward__c WHERE Campaign_Offer__r.Campaign__c=:campId ];
+		
+		for(CW_Audience_Group_Schedule__c rec: [SELECT Id,name,Audience_Group__c,Schedule__c FROM CW_Audience_Group_Schedule__c])
```

Comment:
If the table in this query contains more than 50'000 records a runtime exception will be thrown. I suggest to introduce a `WHERE` clause or cap the max number of records returned adding a `LIMIT` clause to the query.